controllers/idler/idler_controller_test.go

```diff
@@ -1324,32 +1330,91 @@ func preparePayloadsSinglePod(t *testing.T, r *Reconciler, namespace, namePrefix
 func preparePayloadCrashloopingPod(t *testing.T, r *Reconciler, namespace, namePrefix string) payloads {
 	standalonePods := make([]*corev1.Pod, 0, 1)
 	startTime := metav1.Now()
+	replicas := int32(3)
 	// Create a standalone pod with no owner which has restart count sum > 50
 	pod := &corev1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      fmt.Sprintf("%s%s-pod-fail", namePrefix, namespace),
 			Namespace: namespace,
 		},
 		Status: corev1.PodStatus{StartTime: &startTime, ContainerStatuses: []corev1.ContainerStatus{
-			{RestartCount: 30},
+			{RestartCount: 52},
 			{RestartCount: 24},
 		}},
 	}
 	err := r.AllNamespacesClient.Create(context.TODO(), pod)
 	require.NoError(t, err)
 	standalonePods = append(standalonePods, pod)
+	// Deployment
+	d := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("%s%s-deployment", namePrefix, namespace), Namespace: namespace},
+		Spec:       appsv1.DeploymentSpec{Replicas: &replicas},
+	}
+	err = r.AllNamespacesClient.Create(context.TODO(), d)
+	require.NoError(t, err)
+	rs := &appsv1.ReplicaSet{
+		ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("%s-replicaset", d.Name), Namespace: namespace},
+		Spec:       appsv1.ReplicaSetSpec{Replicas: &replicas},
+	}
+	err = controllerutil.SetControllerReference(d, rs, r.Scheme)
+	require.NoError(t, err)
+	err = r.AllNamespacesClient.Create(context.TODO(), rs)
+	require.NoError(t, err)
+	controlledPods := createPods(t, r, rs, startTime, make([]*corev1.Pod, 0, 3), true)
+
+	allPods := append(standalonePods, controlledPods...)
 	return payloads{
 		standalonePods: standalonePods,
-		allPods:        standalonePods,
+		allPods:        allPods,
+		controlledPods: controlledPods,
+		deployment:     d,
 	}
 }
 
-func createPods(t *testing.T, r *Reconciler, owner metav1.Object, startTime metav1.Time, podsToTrack []*corev1.Pod, conditions ...corev1.PodCondition) []*corev1.Pod {
+func preparePayloadCrashloopingPodsWithinThreshold(t *testing.T, r *Reconciler, namespace, namePrefix string, times payloadStartTimes) payloads {
+	startTime := metav1.NewTime(times.defaultStartTime)
+	replicas := int32(3)
+	controlledPods := make([]*corev1.Pod, 0, 3)
+	// Create a StatefulSet with Crashlooping pods less than threshold
+	sts := &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("%s%s-statefulset", namePrefix, namespace), Namespace: namespace},
+		Spec:       appsv1.StatefulSetSpec{Replicas: &replicas},
+	}
+	err := r.AllNamespacesClient.Create(context.TODO(), sts)
+	require.NoError(t, err)
+	for i := 0; i < int(replicas); i++ {
+		pod := &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("%s-pod-%d", sts.Name, i), Namespace: sts.Namespace},
+			Status: corev1.PodStatus{StartTime: &startTime, ContainerStatuses: []corev1.ContainerStatus{
+				{RestartCount: 30},
```

Comment:
maybe worth moving all replica count numbers to consts in case we want to change the threshold later. It will be easier to update the tests.

Reply:
done in [82f8118](https://github.com/codeready-toolchain/member-operator/pull/630/commits/82f8118a99f81e4c34d39486a26b46c419193af3)